### PR TITLE
[Translation][Writer] avoid calling setBackup if the dumper is not FileDumper

### PR DESCRIPTION
**Pull request by aitboudad**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | ~ |
| License | MIT |
| Doc PR | n/a |

### CODE REVIEW

File: src/Symfony/Component/Translation/Writer/TranslationWriter.php [hunk 1]
```diff
@@ -45,7 +46,9 @@ public function addDumper($format, DumperInterface $dumper)
     public function disableBackup()
     {
         foreach ($this->dumpers as $dumper) {
-            $dumper->setBackup(false);
+            if ($dumper instanceof FileDumper) {
```
Comment (xabbuh) [inline src/Symfony/Component/Translation/Writer/TranslationWriter.php]: What about using `method_exists()` here instead to support users who create their own dumpers?

Comment (aitboudad) [inline src/Symfony/Component/Translation/Writer/TranslationWriter.php]: done

**Comment by xabbuh** [comment]
:+1: 

Status: Reviewed

File: src/Symfony/Component/Translation/Writer/TranslationWriter.php [hunk 2]
```diff
@@ -13,6 +13,7 @@
 
 use Symfony\Component\Translation\MessageCatalogue;
 use Symfony\Component\Translation\Dumper\DumperInterface;
+use Symfony\Component\Translation\Dumper\FileDumper;
```
Comment (xabbuh) [inline src/Symfony/Component/Translation/Writer/TranslationWriter.php]: This should now be removed again.

Comment (aitboudad) [inline src/Symfony/Component/Translation/Writer/TranslationWriter.php]: fixed thanks!

File: src/Symfony/Component/Translation/Tests/Writer/TranslationWriterTest.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    public function testWriteTranslations()
+    {
+        $dumper = $this->getMock('Symfony\Component\Translation\Dumper\DumperInterface');
+        $dumper
+            ->expects($this->once())
+            ->method('dump');
+
+        $writer = new TranslationWriter();
+        $writer->addDumper('test', $dumper);
+        $writer->writeTranslations(new MessageCatalogue([]), 'test');
```
Comment (xabbuh) [inline src/Symfony/Component/Translation/Tests/Writer/TranslationWriterTest.php]: Oh, I missed that but tests are failing: We cannot use the short array syntax here.

**Comment by xabbuh** [comment]
Status: Needs work

File: src/Symfony/Component/Translation/Tests/Writer/TranslationWriterTest.php (hunk 3 above)
Comment (aitboudad) [inline src/Symfony/Component/Translation/Tests/Writer/TranslationWriterTest.php]: it should work now, I'll pay more attention next time ;)

**Comment by nicolas-grekas** [comment]
Would it make sense to create an interface for setBackup on 3.1? method_exists is quite fragile imho

**Comment by fabpot** [comment]
Thank you @aitboudad.

**Pull request merged**